### PR TITLE
rather than using the commit, use the diff to master

### LINKED ISSUE #5007: monorepo.sh: CI should build if diff from master (vs previous commit)
**Is your feature request related to a problem? Please describe.**
The current configuration for `monorepo.sh` will skip a CI pass if there were no changes in the commit.  The trouble is CI also automatically cancels builds when another commit is added to a PR.  This can lead to completely missing a test pass that you'd expect was included.  It's easy to miss, looking at the CI build times is the easiest way to know this might have happened.

**Describe the solution you'd like**
Instead of checking if the commit has changes, check if the diff vs master has changes.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by julien51**
# Description

this should let us detect changes when there are multiple commits in a branch.

Fixes #5007

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [X] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
     steps:
       - checkout
       - run:
-          command: scripts/monorepo.sh unlockprotocol $CIRCLE_SHA1 $CIRCLE_BRANCH
+          command: scripts/monorepo.sh unlock-protocol.com $CIRCLE_BRANCH
```
Comment (julien51) [inline .circleci/config.yml]: `unlockprotocol` was never matching anything!

Comment (cnasc) [reply .circleci/config.yml]: Good catch!

**Comment by cnasc** [comment]
> This will be slightly less performant because if the currently pushed branch is not rebased against master, t may run test which would not be required. Maybe a later improvement would be to diff against the earliest diverging commit?

I'd wager that most of our PRs have a small diff against master, though as a best-practices sort of thing we should definitely periodically rebase the long-running PRs to keep them roughly up to date.

**Comment by julien51** [comment]
Dammed, killinhg the job made tests fails. I'll rereun!

**Pull request merged**